startup_scripts/230_virtual_machines.py

```diff
@@ -0,0 +1,65 @@
+from dcim.models import Site, Platform, DeviceRole
+from virtualization.models import Cluster, VirtualMachine
+from virtualization.constants import VM_STATUS_CHOICES
+from tenancy.models import Tenant
+from extras.models import CustomField, CustomFieldValue
+from ruamel.yaml import YAML
+
+from pathlib import Path
+import sys
+
+file = Path('/opt/netbox/initializers/virtual_machines.yml')
+if not file.is_file():
+  sys.exit()
+
+with file.open('r') as stream:
+  yaml = YAML(typ='safe')
+  virtual_machines = yaml.load(stream)
+
+  required_assocs = {
+    'cluster': (Cluster, 'name')
+  }
+
+  optional_assocs = {
+    'tenant': (Tenant, 'name'),
+    'platform': (Platform, 'name'),
+    'role': (DeviceRole, 'name')
+  }
+
+  if virtual_machines is not None:
+    for params in virtual_machines:
+      custom_fields = params.pop('custom_fields', None)
+
+      for assoc, details in required_assocs.items():
+        model, field = details
+        query = { field: params.pop(assoc) }
+
+        params[assoc] = model.objects.get(**query)
+
+      for assoc, details in optional_assocs.items():
+        if assoc in params:
+          model, field = details
+          query = { field: params.pop(assoc) }
+
+          params[assoc] = model.objects.get(**query)
+
+      if 'status' in params:
+        for vm_status in VM_STATUS_CHOICES:
+          if params['status'] in vm_status:
+            params['status'] = vm_status[0]
```

Comment:
Sure, I didn't add it because in the example you sent it wasn't done https://github.com/netbox-community/netbox-docker/blob/master/startup_scripts/110_devices.py#L52-L55 , since those lists are like 3 items each I thought you didn't want to use breaks. Now it's there.